README.md
openmediavault-domoticz
=======================


debian/changelog
openmediavault-domoticz (3.1) stable; urgency=low

  * Add ability to change HTTP and HTTPS ports

 -- OpenMediaVault Plugin Developers <plugins@omv-extras.org>  Fri, 22 Sep 2017 06:21:09 -0500

openmediavault-domoticz (3.0) stable; urgency=low

  * initial release

 -- OpenMediaVault Plugin Developers <plugins@omv-extras.org>  Wed, 01 Feb 2017 18:24:13 -0600


usr/share/openmediavault/confdb/create.d/conf.service.domoticz.sh
#!/bin/sh
#
# @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
# @author    Volker Theile <volker.theile@openmediavault.org>
# @author    OpenMediaVault Plugin Developers <plugins@omv-extras.org>
# @copyright Copyright (c) 2009-2013 Volker Theile
# @copyright Copyright (c) 2013-2017 OpenMediaVault Plugin Developers
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program. If not, see <http://www.gnu.org/licenses/>.

set -e

. /usr/share/openmediavault/scripts/helper-functions

SERVICE_XPATH_NAME="domoticz"
SERVICE_XPATH="/config/services/${SERVICE_XPATH_NAME}"

if ! omv_config_exists "${SERVICE_XPATH}"; then
    omv_config_add_node "/config/services" "${SERVICE_XPATH_NAME}"
    omv_config_add_key "${SERVICE_XPATH}" "enable" "0"
    omv_config_add_key "${SERVICE_XPATH}" "httpenable" "1"
    omv_config_add_key "${SERVICE_XPATH}" "httpport" "8077"
    omv_config_add_key "${SERVICE_XPATH}" "httpsenable" "1"
    omv_config_add_key "${SERVICE_XPATH}" "httpsport" "8078"
fi

exit 0


var/www/openmediavault/js/omv/module/admin/service/domoticz/Domoticz.js
/**
 * Copyright (C) 2015-2017 OpenMediaVault Plugin Developers
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
// require("js/omv/WorkspaceManager.js")

OMV.WorkspaceManager.registerNode({
    id      : "domoticz",
    path    : "/service",
    text    : _("Domoticz"),
    icon16  : "images/snmp.png",
    iconSvg : "images/snmp.svg"
});


var/www/openmediavault/js/omv/module/admin/service/domoticz/Settings.js
/**
 * Copyright (C) 2015-2017 OpenMediaVault Plugin Developers
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

// require("js/omv/WorkspaceManager.js")
// require("js/omv/workspace/form/Panel.js")

Ext.define("OMV.module.admin.service.domoticz.Settings", {
    extend : "OMV.workspace.form.Panel",

    rpcService   : "Domoticz",
    rpcGetMethod : "getSettings",
    rpcSetMethod : "setSettings",

    getFormItems : function() {
        var me = this;
        return [{
            xtype    : "fieldset",
            title    : "General settings",
            defaults : {
                labelSeparator : ""
            },
            items : [{
                xtype      : "checkbox",
                name       : "enable",
                fieldLabel : _("Enable"),
                checked    : false
            }]
        },{
            xtype    : "fieldset",
            title    : "HTTP settings",
            defaults : {
                labelSeparator : ""
            },
            items : [{
                xtype      : "checkbox",
                name       : "httpenable",
                fieldLabel : _("HTTP Enable"),
                checked    : true,
                listeners: {
                    scope: me,
                    change: function(field) {
                        if (field.value == false) {
                            var https = me.findField("httpsenable");
                            https.setValue(true);
                        }
                    }
                }
            },{
                xtype: 'numberfield',
                name: 'httpport',
                fieldLabel: _('HTTP Port'),
                minValue: 8000,
                maxValue: 10000,
                allowDecimals: false,
                allowBlank: true
            }]
        },{
            xtype    : "fieldset",
            title    : "HTTPS settings",
            defaults : {
                labelSeparator : ""
            },
            items : [{
                xtype      : "checkbox",
                name       : "httpsenable",
                fieldLabel : _("HTTPS Enable"),
                checked    : true,
                listeners: {
                    scope: me,
                    change: function(field) {
                        if (field.value == false) {
                            var http = me.findField("httpenable");
                            http.setValue(true);
                        }
                    }
                }
            },{
                xtype: 'numberfield',
                name: 'httpsport',
                fieldLabel: _('HTTPS Port'),
                minValue: 8000,
                maxValue: 10000,
                allowDecimals: false,
                allowBlank: true
            }]
        }];
    }
});

OMV.WorkspaceManager.registerPanel({
    id        : "settings",
    path      : "/service/domoticz",
    text      : _("Settings"),
    position  : 10,
    className : "OMV.module.admin.service.domoticz.Settings"
});


usr/share/openmediavault/confdb/migrations.d/conf.service.domoticz_3.1.sh
#!/bin/sh
#
# @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
# @author    Volker Theile <volker.theile@openmediavault.org>
# @author    OpenMediaVault Plugin Developers <plugins@omv-extras.org>
# @copyright Copyright (c) 2009-2013 Volker Theile
# @copyright Copyright (c) 2013-2017 OpenMediaVault Plugin Developers
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program. If not, see <http://www.gnu.org/licenses/>.

set -e

. /usr/share/openmediavault/scripts/helper-functions

SERVICE_XPATH_NAME="domoticz"
SERVICE_XPATH="/config/services/${SERVICE_XPATH_NAME}"

if ! omv_config_exists "${SERVICE_XPATH}/httpenable"; then
    omv_config_add_key "${SERVICE_XPATH}" "httpenable" "1"
fi

if ! omv_config_exists "${SERVICE_XPATH}/httpport"; then
    omv_config_add_key "${SERVICE_XPATH}" "httpport" "8077"
fi

if ! omv_config_exists "${SERVICE_XPATH}/httpsenable"; then
    omv_config_add_key "${SERVICE_XPATH}" "httpsenable" "1"
fi

if ! omv_config_exists "${SERVICE_XPATH}/httpsport"; then
    omv_config_add_key "${SERVICE_XPATH}" "httpsport" "8078"
fi

exit 0
